src/api/archive_unstable_storage.md

```diff
@@ -3,91 +3,119 @@
-    "params": {
-        "subscription": "...",
-        "result": ...
-    }
+    "result": [
+        {
+            ...
+        },
+        ...
+    ],
+    "discardedItems": ...
 }
 ```
 
-Where `subscription` is equal to the value returned by this function, and `result` can be one of:
+Where:
+
+- `result` contains a vector of JSON objects (possibly empty) that were found in the storage.
+- `discardedItems` is an integer indicating the number of items at the back of the array of the `items` parameters that couldn't be processed.
+
+The JSON objects in the `"result"` field can have one of the following formats based on their type:
 
-### item
+### Value
 
-```json
+```
 {
-    "event": "item",
     "key": "0x0000000...",
     "value": "0x0000000...",
 }
 ```
 
-Yields an item that was found in the storage.
+The JSON object corresponds to one of the requested items whose `type` was `"value"` or `"descendantsValues"`.
+
+If the `key` is not associated with a storage value in the trie, then no response is generated.
+
+Returned when the `type` of the query was `"value"`:
+
+- `key` is guaranteed to be equal to one of the `key`s provided.
+- `value` is a string containing the hexadecimal-encoded value of the storage entry.
 
-The `key` field is a string containing the hexadecimal-encoded key of the value that was found.
-If the `includeDescendants` parameter was `true`, this `key` is guaranteed to start with the `key` provided as parameter.
-If the `includeDescendants` parameter was `false`, then it is also guaranteed to be equal to the `key` provided as parameter.
+Returned when the `type` of the query was `"descendantsValues"`:
 
-The `value` field is a string containing the hexadecimal-encoded value of the storage item.
+- `key` is guaranteed to start with one of the `key`s provided.
+- `value` is a string containing the hexadecimal-encoded value of the storage entry.
 
-### waiting-for-continue
+### Hash
 
-```json
+```
 {
-    "event": "waiting-for-continue"
+    "key": "0x0000000...",
+    "hash": "0x0000000...",
 }
 ```
 
-The `waiting-for-continue` event is generated after at least one `"item"` event has been generated, and indicates that the JSON-RPC client must call `archive_unstable_storageContinue` before more events are generated.
+The JSON object corresponds to one of the requested items whose `type` was `"hash"` or `"descendantsHashes"`.
+
+If the `key` is not associated with a storage value in the trie, then no response is generated.
 
-This event only ever happens if the `includeDescendants` parameter was `true`.
+Returned when the `type` of the query was `"hash"`:
 
-While the JSON-RPC server is waiting for `archive_unstable_storageContinue` to be called, it can generate a `stop` event indicating that it can no longer proceed with that storage access.
+- `key` is guaranteed to be equal to one of the `key`s provided.
+- `hash` is a string containing the hexadecimal-encoded cryptographic hash of the storage entry.
```

Comment:
picking hairs, `twox` is not a cryptographic hash function and it seems unnecessary to specify that here.

```suggestion
- `hash` is a string containing the hexadecimal-encoded hash of the storage entry.
```